storagepool/agent.go

```diff
@@ -17,20 +17,22 @@ type StoragepoolAgent struct {
 	storagepoolRootDir  string
 	storagepoolUuid     string
 	healthCheckBaseDir  string
+	healthCheckType     string
 	cattleClient        cattle.CattleInterface
 }
 
-func NewStoragepoolAgent(healthCheckInterval int, storagepoolRootDir, storagepoolUuid, healthCheckBaseDir string, cattleClient cattle.CattleInterface) *StoragepoolAgent {
+func NewStoragepoolAgent(healthCheckInterval int, storagepoolRootDir, storagepoolUuid, healthCheckBaseDir, healthCheckType string, cattleClient cattle.CattleInterface) *StoragepoolAgent {
```

Comment:
Just pass in the health checker object, not the type.  the caller can construct the health checker.